Content/Items/Forest/Armors.SlimePrinceMinion.cs

```diff
@@ -0,0 +1,357 @@
+				Projectile.timeLeft = 2;
+
+			if (Vector2.Distance(Projectile.Center, Owner.Center) > 5000)
+				Projectile.Center = Owner.Center;
+
+			switch (State)
+			{
+				case 0: PassiveBehavior(); break;
+				case 1: AttackBehavior(); break;
+				case 2: FuseAnimation(); break;
+				case 3: FusedBehavior(); break;
+			}
+		}
+
+		/// <summary>
+		/// Behavior this minion should follow when they are idle, following the player
+		/// and trying to keep up with them.
+		/// </summary>
+		public void PassiveBehavior()
+		{
+			if (HealTimer % 2 == 0 && life < MAX_LIFE)
+				life++;
+
+			Vector2 targetPos = Owner.Center + new Vector2(32 * Owner.direction * -1, -32);
+
+			Projectile.velocity = (targetPos - Projectile.Center) * 0.1f;
+
+			if (Projectile.velocity.Length() > 15)
+				Projectile.velocity = Vector2.Normalize(Projectile.velocity) * 14.99f;
+
+			Projectile.frame = Timer % 40 < 20 ? 1 : 0;
+
+			// Check target
+			target = Projectile.TargetStrongestNPC(800, true, true);
+
+			// Switch to attack mode if applicable
+			if (target != null)
+			{
+				State = 1;
+				Timer = 0;
+			}
+		}
+
+		/// <summary>
+		/// Behavior the minion should follow when it has an active, valid target and is 
+		/// attacking them passively.
+		/// </summary>
+		public void AttackBehavior()
+		{
+			if (HealTimer % 2 == 0 && life < MAX_LIFE)
+				life++;
+
+			// If we need to retarget, do so and reset the attack
+			if (!Projectile.TargetValid(800, true, true, target))
+			{
+				target = Projectile.TargetStrongestNPC(800, true, true);
+				Timer = 0;
+			}
+
+			// return to idle if no target
+			if (target is null)
+			{
+				State = 0;
+				Timer = 0;
+				return;
+			}
+
+			// Otherwise, we follow the attack cycle
+			if (Timer == 1)
+			{
+				startPoint = Projectile.Center;
+			}
+
+			// Position above the target
+			if (Timer > 1 && Timer < 30)
+			{
+				Projectile.velocity *= 0;
+				Vector2 targetPos = target.Center + Vector2.UnitY * (target.height * -0.5f - 64);
+				Projectile.Center = Vector2.SmoothStep(startPoint, targetPos, Timer / 30f);
+			}
+
+			// Fall onto them
+			if (Timer > 30 && Timer < 90)
+			{
+				Projectile.velocity.Y += 0.4f;
+				Projectile.Center = new Vector2(target.Center.X, Projectile.Center.Y);
+				Projectile.frame = 1;
+
+				// Bounce off the enemy once it's been stomped on
+				if (Projectile.Hitbox.Intersects(target.Hitbox))
+				{
+					Projectile.velocity.Y = -7;
+					Projectile.velocity.X = Main.rand.NextFloat(-5, 5);
+					Timer = 90;
+				}
+			}
+
+			// Continue to experience gravity, albeit reduced
+			if (Timer >= 90)
+			{
+				Projectile.velocity.Y += 0.2f;
+				Projectile.frame = 0;
+			}
+
+			// Fire a thorn at them
+			if (Timer == 110)
+			{
+				Projectile.NewProjectile(Projectile.GetSource_FromThis(), Projectile.Center, Projectile.Center.DirectionTo(target.Center + target.velocity * 10) * 11, ModContent.ProjectileType<SlimeThorn>(), 20, 0.5f, Projectile.owner);
```

Comment:
Projectile.newProjectile needs to run for only one client or the server but not all otherwise it'll duplicate